### PR TITLE
docs/provider: Update Contributing Guide to elide HashiCorp vs. Community Providers section, add links to GitHub Help, and add Go Coding Style references

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

The HashiCorp vs. Community Providers section of the contributing guide has been passed down from when the main hashicorp/terraform repository contained all Terraform Providers. Since the Terraform 0.10 split and the formation of the terraform-providers GitHub organization with individual Terraform Provider repositories, this documentation is extraneous and potentially confusing for contributors.

The additional links in the Pull Request Lifecycle section are designed to provide some extra assistance to those not familiar with GitHub. This also switches the recommendation for "not quite ready" pull requests from "[WIP]" in the pull request title to GitHub's draft pull request feature.

The final update of adding the Go Coding Style section under the Common Review Items section is to provide additional references of stylistic preferences within the codebase, especially for contributors from other programming languages and where our linting tooling does not provide automated recommendations.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing: N/A documentation update

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -153,27 +121,28 @@ expect:
...
+   You are welcome to submit your pull request for commentary or review before
+   it is fully completed by creating a [draft pull request](https://help.github.com/en/articles/about-pull-requests#draft-pull-requests)
+   and including specific questions or items you'd like feedback on.
 
-2. Once you believe your pull request is ready to be merged, you can remove the
-   "[WIP]" prefix from the title and a team member will review. Follow [the
-   checklists below](#checklists-for-contribution) to help ensure that your
-   contribution will be merged quickly.
+1. Once you believe your pull request is ready to be reviewed, ensure the
+   pull request is not a draft pull request, and a team member will review. Follow
```
Comment (ryndaniels) [inline .github/CONTRIBUTING.md]: Is there a guide on how to make sure it's not a draft PR? (https://help.github.com/en/articles/changing-the-stage-of-a-pull-request maybe?)

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -333,6 +302,13 @@ quality. For any guidelines listed, contributors are encouraged to ask any
 questions and community reviewers are encouraged to provide review suggestions
 based on these guidelines to speed up the review and merge process.
 
+#### Go Coding Style
```
Comment (ryndaniels) [inline .github/CONTRIBUTING.md]: 😻

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (nywilken) [reply .github/CONTRIBUTING.md]: Can draft PRs be created outside of the GitHub Frontend? If not, I think we should still encourage the use of WIP PRs if draft PRs is not something contributors are use to or if their tooling is not capable of submitting/ creating.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -333,6 +302,13 @@ quality. For any guidelines listed, contributors are encouraged to ask any
...
+
+The following Go language resources provide common coding preferences that may be referenced during review, if not automatically handled by the project's linting tools.
+
+- [Effective Go](https://golang.org/doc/effective_go.html)
+- [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
```
Comment (nywilken) [inline .github/CONTRIBUTING.md]: :+1:

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (bflad) [reply .github/CONTRIBUTING.md]: I've hopefully addressed both of these in https://github.com/terraform-providers/terraform-provider-aws/pull/9031/commits/133dcbf0b65fee71eca25588c3c91621acd1a8db 😄 Thanks for the recommendations!

**Pull request merged**